include/openPMD/RecordComponent.hpp

```diff
@@ -78,6 +78,40 @@ struct IsContiguousContainer< std::array< T_Value, N > >
 template< typename T >
 class DynamicMemoryView;
 
+class RecordComponent;
+
+namespace internal
+{
+    class RecordComponentData : public BaseRecordComponentData
+    {
+        friend class openPMD::RecordComponent;
+
+        RecordComponentData();
+
+        RecordComponentData( RecordComponentData const & ) = delete;
+        RecordComponentData( RecordComponentData && ) = delete;
+
+        RecordComponentData & operator=( RecordComponentData const & ) = delete;
+        RecordComponentData & operator=( RecordComponentData && ) = delete;
+
+        std::queue< IOTask > m_chunks;
```

Comment:
For all the `...Data` classes, it would be really good if we try to catch up and add Doxygen strings to the `m_` member variables.

Can be a follow-up and we can split the work if we are unsure about some. I remember we found them hard to follow at times and it's good to document the purpose of individual `m_` member variables.

Reply:
Note: Inside #1154 I began aliasing these data classes (needed that for something else in there) which made the whole thing a lot more readable in my opinion, e.g.:
```cpp
namespace internal {
    class RecordComponentData{ /* … */ };
}

class RecordComponent {
    using DataClass = internal::RecordComponentData;
    std::shared_ptr< DataClass > m_recordComponentData;
    /* … */
};
```

Reply:
That is very nice, yes :sparkling_heart: 

But my comment was actually about adding this to the members in the first class, so we know what they are for:
```C++
struct RecordComponentData
{
    // ...

    //! container for holding data chunks
    std::queue< IOTask > m_chunks;

    //! this sets things on fire
    bool m_fire;
};
```

Reply:
Ah, got you
Yep, these classes are probably a good chance to have documentation for our internal state at one central place